packages/menu/src/Menu/Menu.tsx

```diff
@@ -241,9 +245,16 @@ export const Menu = React.forwardRef<HTMLDivElement, MenuProps>(function Menu(
       event?.nativeEvent?.stopPropagation?.();
     };
 
+    const triggerKeyDownHandler = (event?: React.KeyboardEvent) => {
+      if (event?.key === keyMap.Enter || event?.key === keyMap.Space) {
+        keyboardUsedRef.current = true;
+      }
+    };
```

Comment:
Trying to decide if hooking into `onKeyDown` like this is better than checking the `event.detail`  in the click handler.
In this case it's clear that we're looking for keydowns, but we've now added another event handler.

Alternatively we could set `keyboardUsedRef` from within the `triggerClickHandler`, but in this case it's not really clear what's going on

```
const triggerClickHandler = (event) => {
  if (event.detail === 0) {
    keyboardUsedRef.current = true;
  }
  /// ...
}
```

(See https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event#usage_notes)

Reply:
If you do decide to keep it as-is, we'll also need to explicitly call `trigger.props.onKeyDown` if it exists